react/react2/week1/github-searcher-app/src/MainComponent.js

```diff
@@ -0,0 +1,57 @@
+import React, { useContext, useEffect, useState, createContext } from "react";
+import "./App.css";
+import { DisplaySearchResult } from "./DisplaySearchResult";
+
+export const FetchedContext = createContext();
+
+export const MainComponent = () => {
+  
+  const [input, setInput] = useState("");
+  const [queryResult, setQueryResult] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const [errMessage, setErrMessage] = useState("");
+
+  const API = `https://api.github.com/search/users?q=${input}`;
+
+  useEffect(() => {
+    
+    if (!input) {
+      setQueryResult([]);
+    } else {
+      setLoading(true);
+      fetch(API)
+        .then((res) => {
+          if (res.ok) {
+            return res.json();
+          }
+          setErrMessage("Error fetching! Error code: " + res.status);
+          return;
+        })
+        .then((data) => {
+          const result = data.items.map((name) => name.login);
+          setQueryResult(result);
+          setLoading(false);
```

Comment:
You are only setting loading false if the response is ok, but you could also set it to false if you get an error and display some error message to the user.